.github/workflows/deploy.yml

```
@@ -0,0 +1,244 @@
+    - name: 📊 Notify deployment
+      run: |
+        echo "📊 Staging deployment notification sent"
+        # Add your notification logic here (Slack, email, etc.)
+
+  deploy-production:
+    name: 🚀 Deploy to Production
+    runs-on: ubuntu-latest
+    needs: [build, test-deployment]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master' || (github.event_name == 'workflow_dispatch' && github.event.inputs.environment == 'production')
+    environment: production
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)